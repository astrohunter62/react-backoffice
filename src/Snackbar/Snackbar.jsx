import React from 'react'
import PropTypes from 'prop-types'

import MaterialSnackbar from 'material-ui/Snackbar'

const Snackbar = ({
  open,
  message,
}) => (
  <MaterialSnackbar
    anchorOrigin={{
      vertical: 'bottom',
      horizontal: 'left',
    }}
    open={open}
    autoHideDuration={6000}
    SnackbarContentProps={{
      'aria-describedby': 'message-id',
    }}
    message={<span id="message-id">{message}</span>}
  />
)

Snackbar.propTypes = {
  open: PropTypes.bool,
  message: PropTypes.string.isRequired,
}

Snackbar.defaultProps = {
  open: false,
}

export default Snackbar
