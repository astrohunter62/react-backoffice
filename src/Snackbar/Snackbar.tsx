import React from "react";
import { Snackbar as MaterialSnackbar } from "@material-ui/core";

type SnackbarProps = {
  isOpen?: boolean;
  message: string;
};

const Snackbar: React.SFC<SnackbarProps> = ({ isOpen, message }) => (
  <MaterialSnackbar
    anchorOrigin={{
      vertical: "bottom",
      horizontal: "left",
    }}
    open={isOpen}
    autoHideDuration={6000}
    ContentProps={{
      "aria-describedby": "message-id",
    }}
    message={<span id="message-id">{message}</span>}
  />
);

Snackbar.defaultProps = {
  isOpen: false,
};

export default Snackbar;
