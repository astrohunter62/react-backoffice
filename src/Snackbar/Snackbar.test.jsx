import React from 'react'
import { shallow } from 'enzyme'

import Snackbar from '.'

it('renders correctly', () => {
  const tree = shallow(<Snackbar message="Message" />)

  expect(tree).toMatchSnapshot()
})
