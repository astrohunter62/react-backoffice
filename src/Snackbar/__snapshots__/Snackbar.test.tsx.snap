// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<WithStyles(ForwardRef(Snackbar))
  ContentProps={
    Object {
      "aria-describedby": "message-id",
    }
  }
  anchorOrigin={
    Object {
      "horizontal": "left",
      "vertical": "bottom",
    }
  }
  autoHideDuration={6000}
  message={
    <span
      id="message-id"
    >
      Message
    </span>
  }
  open={false}
/>
`;
